File: backend/models/site.py
from config.db import db, app, ma

class Site(db.Model):
    __tablename__ = 'sites'

    id = db.Column(db.Integer, primary_key=True)
    nombre = db.Column(db.String(100), nullable=False)
    categoria = db.Column(db.String(50), nullable=False)
    descripcion = db.Column(db.Text, nullable=False)
    ubicacion = db.Column(db.String(100), nullable=False)
    abierto = db.Column(db.Boolean, default=True)

    def __init__(self, nombre, categoria, descripcion, ubicacion, abierto):
        self.nombre = nombre
        self.categoria = categoria
        self.descripcion = descripcion
        self.ubicacion = ubicacion
        self.abierto = abierto

with app.app_context():
    db.create_all()

class SiteSchema(ma.Schema):
    class Meta:
        fields = ('id', 'nombre', 'categoria', 'descripcion', 'ubicacion', 'abierto')

File: lib/models/site.dart
class Site {
  final int id;
  final String nombre;
  final String categoria;
  final String descripcion;
  final String ubicacion;
  final bool abierto;

  Site({required this.id, required this.nombre, required this.categoria, required this.descripcion, required this.ubicacion, required this.abierto});

  factory Site.fromJson(Map<String, dynamic> json) {
    return Site(
        id: json['id'],
        nombre: json['nombre'],
        categoria: json['categoria'],
        descripcion: json['descripcion'],
        ubicacion: json['ubicacion'],
        abierto: json['abierto'],
    );
  }
}

